### PR TITLE
16 list command

### PR DESCRIPTION
**Pull request by unbekanntunity**
Changes
- Add list command

Task:
"------------"(without "")
Task Id: (1)
Description: Just a example
Status: Started
"------------"
and so on....

### CODE REVIEW

File: BusyList/Handlers/DoneHandler.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using BusyList.Commands;
+using System;
+
+namespace BusyList.Handlers
+{
+    public class DoneHandler : IHandler<DoneCommand>
```
Comment (pobiega) [inline BusyList/Handlers/DoneHandler.cs:6]: Looks like you based this branch on DoneHandler instead of Master. You'll need to rebase it, I'm afraid.

Comment (pobiega) [reply BusyList/Handlers/DoneHandler.cs:6]: Or set it to blocked by #20, that works too - as you need the TaskStatus property.

File: BusyList/Handlers/ListHandler.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public ListHandler(ITaskRepository taskRepository)
+        {
+            _taskRepository = taskRepository;
+        }
+
+        public void Run(DoneCommand command)
+        {
+            IEnumerable<TaskItem> tasks = _taskRepository.GetAll();
+
+            string seperator = new String('-', 10);
```
Comment (pobiega) [inline BusyList/Handlers/ListHandler.cs:20]: Typo: separator.

File: BusyList/Handlers/ListHandler.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        private readonly ITaskRepository _taskRepository;
+
+        public ListHandler(ITaskRepository taskRepository)
+        {
+            _taskRepository = taskRepository;
+        }
+
+        public void Run(DoneCommand command)
+        {
+            IEnumerable<TaskItem> tasks = _taskRepository.GetAll();
```
Comment (pobiega) [inline BusyList/Handlers/ListHandler.cs:18]: Use `var` please.

File: BusyList/Handlers/ListHandler.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+            Console.WriteLine("Tasks:");
+
+            foreach (var item in tasks)
+            {
+                Console.WriteLine(seperator);
+                Console.WriteLine($"Task id: {item.Id}");
+                Console.WriteLine($"Description: {item.Description}");
+                Console.WriteLine($"Status: {item.TaskStatus}");
+                Console.WriteLine(seperator);
```
Comment (pobiega) [inline BusyList/Handlers/ListHandler.cs:30]: The separator will look a bit weird if there are multiple items:

"---------"
text
"---------"
"---------"
text
"---------"

To fix this, I suggest moving the first separator print before the loop, which will create the "expected" output in all cases.

**Review by pobiega — CHANGES_REQUESTED** [review]

File: BusyList/Handlers/ListHandler.cs (hunk 2 above)
Comment (pobiega) [inline BusyList/Handlers/ListHandler.cs:20]: This should be constant.

**Review by pobiega — CHANGES_REQUESTED** [review]

File: BusyList/Handlers/DoneHandler.cs (hunk 1 above)
Comment (unbekanntunity) [reply BusyList/Handlers/DoneHandler.cs:6]: I based it on master and how I can block it?

**Review by pobiega — APPROVED** [review]

**Review by chrisK00 — APPROVED** [review]

**Pull request merged**